docs/API.md

```diff
@@ -51,19 +80,34 @@ Empty for now.
 - `Operation.type()` : string - Returns either `ClientSubscribing`, `ClientPublishing`, `ApplicationSubscribing`, `ApplicationPublishing`
 
 ### Channel
-- `Channel.path()` : string
+- `Channel.path()` : string (outcome of https://github.com/asyncapi/markdown-template/blob/master/components/Channels.js#L26)
 - `Channel.messages()` : Message[]
 - `Channel.operations()` : Operation[]
+- `Channel.hasDescription()` : boolean (outcome of https://github.com/asyncapi/markdown-template/blob/master/components/Channels.js#L28)
 - `Channel.description()` : string (outcome of comment(s): [799481319](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799481319), [800282407](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800282407))
 - `Channel.extension('<extension property>')` : any (outcome of comment(s): [799598596](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799598596))
 - `Channel.binding('<binding protocol>')` : any (outcome of comment(s): [799481319](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799481319), [799598596](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799598596))
-- `Channel.parameters()` : Schema[] (outcome of comment(s): [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961))
+- `Channel.hasParameters()` : boolean (outcome of https://github.com/asyncapi/markdown-template/blob/master/components/Channels.js#L34)
+- `Channel.parameters()` : ChannelParameter[] (outcome of comment(s): [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961) and https://github.com/asyncapi/markdown-template/blob/master/components/Channels.js#L34)
```

Comment:
I have to say I have mixed fillings with proposal for stuff like `Channel.*`.

in my template, when I need parameters, I know:
- I need parameters
- what topic/channel are they part off

so I want `getParameters(channelName)` or `getParameters()` to get all params or `getParameter(paramName, channelName)` or `getParameter(paramName)` to get param from `component.params`

much better experience in IDE imho

Reply:
@derberg in this particular case, `Channel.parameters()` is not an intent from the root object, just a needed getter. Those are needed since users would like to also get parameters from the channel object directly.

It seems what you miss is an intent on the root, such as `asyncApi.channelParameters(<channel_path>) : ChannelParameter[]`.

I see it as a valid intent indeed, however, it would be nice if you can provide where your use case is located (which template?) so we can also evaluate if we could go further and improve this intent.

Reply:
let me take a step back to not block you. I have mixed feeling towards the entire intend API and tbh I won't fix it by just being stubborn on one use case 😄 Just go ahead 👍🏼